### ISSUE TITLE
Help on `FastifyError: Reply was already sent` error

### ISSUE THREAD

**Issue by bacloud22** [fastify/help#718]
## 💬  I'm having this error occasionally: 
First, I believe I did never spot this error before because I removed an error handler that I was using and have dealt with each error independently.

> {"level":40,"time":1657952480291,"pid":19,"hostname":"61966507051d","reqId":"req-a9","err":{"type":"FastifyError","message":"Reply was already sent.","stack":"FastifyError: Reply was already sent.\n    at _Reply.Reply.send (/classified-ads/node_modules/fastify/lib/reply.js:113:26)\n    at /classified-ads/node_modules/@fastify/view/index.js:393:14\n    at getTemplate (/classified-ads/node_modules/@fastify/view/index.js:177:7)\n    at _Reply.viewEjs (/classified-ads/node_modules/@fastify/view/index.js:369:5)\n    at _Reply.view (/classified-ads/node_modules/@fastify/view/index.js:120:14)\n    at _Reply.blabla (file:///classified-ads/libs/decorators/blabla.js:61:14)\n    at Object.<anonymous> (file:///classified-ads/libs/routes/index.js:33:26)\n    at runMicrotasks (<anonymous>)\n    at processTicksAndRejections (internal/process/task_queues.js:95:5)","name":"FastifyError","code":"FST_ERR_REP_ALREADY_SENT","statusCode":500},"msg":"Reply already sent"}

My route is like:
```js
    fastify.get('/', { preHandler: softAuth }, async function (req, reply) {
        const [err, listings] = await to(QInstance.getListingsSince(20, '', req.pagination))
        if (err) {
            req.log.error(`index#getListingsSince: ${err.message}`)
            return reply.blabla([{}, 'message', 'SERVER_ERROR'], req)
        }
        // doing stuff
        reply.blabla([data, 'index', 'listings'], req)
        return reply
    })
```

While `reply.blabla` decorator is not asynchronous and producing no errors to my belief. It just do some translations and delivers like:

```js
// Send JSON for API env
if (NODE_ENV == -1) {
    this.send(context[0])
} else {
    this.view(`/templates/pages/${route}`, data)
}
```
I have a lesson I took to heart," _return or await reply on async routes,_",, "_return or await reply on async routes_", but it failed me this time !!

- *node version*: 14
- *fastify version*: >=4.0.0
- *os*: Linux Ubuntu 22

Thanks for your attention

**Comment by mcollina** [comment]
Thanks for reporting!

Can you provide steps to reproduce? We often need a [reproducible example](https://stackoverflow.com/help/minimal-reproducible-example), e.g. some code that allows someone else to recreate your problem by just copying and pasting it. If it involves more than a couple of different file, create a new repository on GitHub and add a link to that.

**Comment by bacloud22** [comment]
hello @mcollina 
I will work on that. Thanks.

**Comment by submgr** [comment]
did you manage to solve this problem? I have exactly the same thing after migrating from version 3 to 4 :(

**Comment by bacloud22** [comment]
Sadly I couldn't reproduce the error. And I'm sorry I forgot to followup per @mcollina recall. 

All I know is that I was using a `onrequest` handler with so much callbacks. I used that for hitting projecthoneypot.org API. 

So all I did is to remove the whole callback altogether. 

I believe this was something like my handler (I abandoned that same version/ project)

https://github.com/bacloud22/classified-ads/blob/d1b2f674b7a8c661c8881e6aef2f1e274d74ea27/app.js#L211
